src/UnderlineNav2/UnderlineNavItem.tsx

```diff
@@ -104,70 +93,37 @@ export const UnderlineNavItem = forwardRef(
         setNoIconChildrenWidth({text, width: domRect.width - iconWidthWithMargin})
-
-        // When an item has aria-current !== false while rendering, we should be sure to select it.
-        // It can happen when the page is loaded (selectedLink === undefined)
-        // or if the item is coming out of the menu when there is enough space to show items along with the more menu. (selectedLink.current === null)
-        if (
-          (selectedLink === undefined || selectedLink.current === null) &&
-          Boolean(ariaCurrent) &&
-          ariaCurrent !== 'false'
-        ) {
-          setSelectedLink(ref as RefObject<HTMLElement>)
-        }
-
-        // Only runs when a menu item is selected (swapping the menu item with the list item to keep it visible)
-        if (selectedLinkText === text) {
-          setSelectedLink(ref as RefObject<HTMLElement>)
-          if (typeof onSelect === 'function' && selectEvent !== null) onSelect(selectEvent)
-          setSelectedLinkText('')
-        }
       }
-    }, [
-      ref,
-      ariaCurrent,
-      selectedLink,
-      selectedLinkText,
-      setSelectedLinkText,
-      setSelectedLink,
-      setChildrenWidth,
-      setNoIconChildrenWidth,
-      onSelect,
-      selectEvent,
-    ])
+    }, [ref, setChildrenWidth, setNoIconChildrenWidth])
 
-    const keyPressHandler = React.useCallback(
-      (event: React.KeyboardEvent<HTMLLIElement>) => {
-        if (event.key === ' ' || event.key === 'Enter') {
-          if (!event.defaultPrevented && typeof onSelect === 'function') onSelect(event)
-          if (!event.defaultPrevented && typeof afterSelect === 'function') afterSelect(event)
-          setSelectedLink(ref as RefObject<HTMLElement>)
+    const keyDownHandler = React.useCallback(
+      (event: React.KeyboardEvent<HTMLAnchorElement>) => {
+        if ((event.key === ' ' || event.key === 'Enter') && !event.defaultPrevented && typeof onSelect === 'function') {
+          onSelect(event)
         }
       },
-      [onSelect, afterSelect, ref, setSelectedLink],
+      [onSelect],
     )
     const clickHandler = React.useCallback(
-      (event: React.MouseEvent<HTMLLIElement>) => {
-        if (!event.defaultPrevented) {
-          if (typeof onSelect === 'function') onSelect(event)
-          if (typeof afterSelect === 'function') afterSelect(event)
+      (event: React.MouseEvent<HTMLAnchorElement>) => {
+        if (!event.defaultPrevented && typeof onSelect === 'function') {
+          onSelect(event)
         }
-        setSelectedLink(ref as RefObject<HTMLElement>)
       },
-      [onSelect, afterSelect, ref, setSelectedLink],
+      [onSelect],
     )
 
     return (
       <Box as="li" sx={{display: 'flex', flexDirection: 'column', alignItems: 'center'}}>
-        <Box
+        <Link
+          ref={ref}
           as={Component}
           href={href}
-          onKeyPress={keyPressHandler}
-          onClick={clickHandler}
           aria-current={ariaCurrent}
```

Comment:
I saw some conversation about `aria-current`. It is the preferred method for accessibility, but I think it can be hard to control sometimes (at least that has been our experience so far with NavList). But maybe we can continue using it but handle the state of it somehow with more control? Or offer multiple solutions?

Reply:
>but I think it can be hard to control sometimes (at least that has been our experience so far with NavList)

Thanks for sharing this! I personally have mix feelings about this API and felt validated when you shared the feedback 😅  We changed the prop name from `selected` to `aria-current` to align with `NavList`. But if we already have feedback from NavList, maybe it is a good time to chat about in the Primer API session??